### ISSUE TITLE
Will not compile on OS X

### ISSUE THREAD

**Issue by watson** [e-conomic/graphicsmagick-stream#6]
Since version 2.1.0 this will not compile on my OS X. It fails with the following error:

```
> graphicsmagick-stream@2.1.0 install /Users/watson/code/node_modules/airserver/node_modules/graphicsmagick-stream
> make

mkdir -p bin
gcc src/convert.c src/io.c -o bin/convert -L/usr/local/lib -larchive -O `GraphicsMagickWand-config --cflags --cppflags --ldflags --libs` 
src/convert.c:9:10: fatal error: 'archive.h' file not found
#include <archive.h>
         ^
```

If I go back to v2.0.1, everything works fine

**Comment by gausby** [comment]
Sorry, this is my fault. This is described in the readme.

I know that this is bad form, but we needed this to run on linux systems, so we got the make file working on that platform but unfortunately not on OS X. If you know how to detect OS X, and do logic in make-files, then a pull request would be welcome.

**Comment by jeanfredrik** [comment]
I have the same problem. Is it not possible to compile on OS X at all right now? I've run
```
brew install graphicsmagick --build-from-source
brew install libarchive
```
but when I try build the binary with
```
gcc src/*.c -o bin/convert -L/usr/local/opt/libarchive/lib -I/usr/local/opt/libarchive/include -larchive -O `GraphicsMagickWand-config --cflags --cppflags --ldflags --libs`
```
I get `clang: error: no such file or directory: 'src/*.c'` and `gm` seems to be working anyway, but not the npm package. I get the same error as @watson when I try to add it.

**Comment by watson** [comment]
@gausby maybe it's my imagination, but didn't you comment on this issue previously? :wink:

**Comment by gausby** [comment]
I did. It was about my previous employers need to only run this on Linux, so the make file is setup to only handle this case. I did make it compile on OS X and I have added a comment about this, but my lack of understanding of make-files at the moment, and the lack of operating systems to test on, didn't make me capable of adding an if-sentence that switch the build command for various OS's.

I thought pointing to my note in the readme about building on OSX would solve the problem, but seeing a comment after mine made me realise that something else must have changed, rendering my comment invalid and unneeded.

**Comment by b-i-l-l** [comment]
Installing `libarchive` via brew does not create symbolic links in `/usr/local/` since OS X has a copy of the library. OS X does not include the header file, which is the source of the first error reported in this thread.

In addition, the `Makefile` for this project does not allow setting of `CFLAGS` or `LDFLAGS` to add the necessary paths created by `brew install`.

The module will compile if you manually create symlinks from `/usr/local/opt/libarchive/include/` into `/usr/local/include/` for `archive.h` and `archive_entry.h` along with the equivalent links for `lib`.

**Comment by mikepc** [comment]
I'm experiencing this issue as well on OS X. The last response isn't an answer so much as a 'don't do this because'. Does anyone have a solution to the issue?

**Comment by samdalton** [comment]
After upgrading to El Capitan, following the suggestion from @b-i-l-l I was able to get `v2.0.1` installed with one other additional command.

First, copy archive files 
`sudo ln -s /usr/local/opt/libarchive/include/archive_entry.h /usr/local/include/archive_entry.h`

`sudo ln -s /usr/local/opt/libarchive/include/archive.h /usr/local/include/archive.h`

Then, I had to disable the new System Integrity Protection following these steps https://developer.apple.com/library/prerelease/mac/documentation/Security/Conceptual/System_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html
(Booting into Recovery OS by restarting and then holding Cmd+R until the Apple logo shows)

That enabled this command to run
`sudo ln -s /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/system/libsystem_stats.dylib /usr/lib/system/libsystem_stats.dylib`

Which finally leads to a successful 
`npm install graphicsmagick-stream@2.0.1`

**Comment by madhums** [comment]
Have the same issue, unable to resolve it. Any workarounds? @gausby

**Comment by gausby** [comment]
No. I am sorry everyone. I really have no interest in this project. I am sorry that I can't give you a solution.

I am not using this project anymore, please talk to some of the other maintainers. If somebody wanna fix this and create a pull-request it would be great, but I am not going to look into this because I focus on other stuff.
